File: LeetCode/LC-1572-Matrix-Diagonal-Sum/Solution.py
class Solution:
    def diagonalSum(self, mat: List[List[int]]) -> int:
        n = len(mat)
        i,j=0,n-1
        sum,row = 0,0
        while row < n:
            if i==j :
                sum += mat[row][i]
            else:
                sum+= mat[row][i]
                sum += mat[row][j]
            i+=1
            j-=1
            row+=1
        return sum
        

File: LeetCode/LC-1572-Matrix-Diagonal-Sum/Solution.java
class Solution {
    public int diagonalSum(int[][] mat) {
        int n = mat.length;
        int row = 0 , i = 0 , j = n-1 ,sum = 0;

        while(row < n){
            if(i==j){
                sum+=mat[row][i];
            }
            else{
                sum+=(mat[row][i] + mat[row][j]);
            }
            i++;j--;
            row++;
        }

        return sum;

    }
}
